### ISSUE TITLE
torchlive-cli setup-dev script fails to install openjdk@8 ("openjdk@8: no bottle available!")

### ISSUE THREAD

**Issue by tzpr** [pytorch/live#29]
### Version

0.1.2

### Problem Area

CLI

### Steps to Reproduce

1. Open terminal in macOS (OS version 12.1, M1 chip)
2. Run command: _npx torchlive-cli setup-dev_
3. Script fails in the OpenJDK installation step: "Error: openjdk@8: no bottle available!"

### Expected Results

Expected the JavaSDK installation to succeed.

What is a working way to install the Java SDK? 
Would it be possible to use a newer Java SDK instead of openjdk@8, like openjdk@17?

**Comment by aaronespasa** [comment]
Hi @tzpr , thanks for reporting the issue!

Currently, PyTorch Live does not work with a version of OpenJDK greater than the version 8. However, the version 8 of the Java Platform has not been built natively for Apple Silicon. 

I recommend you to **execute the terminal using Rosetta**:
    - Go to the Applications folder, do right click on the Terminal app and select *Get Info*.
    - Click on the checkbox "Open using Rosetta".
<p align="center"><img src="https://user-images.githubusercontent.com/39239895/148265073-dc4575e8-11ba-4c5c-806f-3ad69407f905.jpeg" alt="Terminal using Rosetta"/></p>

By executing the terminal using Rosetta I think that you won't have any issues at the time of installing openjdk@8 using HomeBrew. I'm using a M1 MacBook and this solution works fine in my case.

Let me know if this solution works also for you!

**Comment by aaronespasa** [comment]
@tzpr If this is the first time that you're installing PyTorch Live using the M1 Chip, I recommend you to install first these dependencies (before executing the `npx torchlive-cli setup-dev` command) executing all the commands with a Terminal that uses Rosetta:

- The **version 3.9.5 of Python** and the command `python3 -V` must output the version 3.9.5. The command `python3` is used to install the requirements and make the models. However, an specific version of torchvision requires a version of Python greater than the version 3.9.0 and smaller than the version 3.9.9 I think. In my case, the version 3.9.5 works fine.
- The version 16 of Node (if you use the version 17 you will run into an issue).
- Watchman: Just to `brew install watchman` (I don't know why this command fails each time I use the command `setup-dev`. So to prevent it, just install it directly using HomeBrew).
- Android Studio and the Android SDK Manager.
- Android Emulator -> Go to Android Studio. If the Android Emulator is not install, you can install it in System Preferences > Appearances & Behavior > System Settings > Android SDK > SDK Tools. If the Android Emulator is already installed, delete it and install it again.
- The PyTorch Live AVD for the ARM architecture. Until the PR #24 is not accepted, I recommend you to do the following steps:

Clone my forked repository and go to the `arm-emulator` branch:
```bash
$ git clone https://github.com/aaronespasa/live.git
$ cd live
$ git checkout arm-emulator
```

Go to the `torchlive-cli` folder and install the Node dependencies:
```bash
$ cd torchlive-cli
$ npm install
```

And now, run the `setup-dev` command from there:
```bash
$ npm run start setup-dev
```

Now, all the dependencies should be correctly installed and you can init your first project and run it using the Android emulator 🎉

If you have some issues during the installation of these dependencies, just let me know and I'll help you as soon as I can!

**Comment by raedle** [comment]
@aaronespasa, thanks a bunch for chiming in and helping out with steps and the PR #24

@tzpr, we merged PR #24. The updates are available in the nightly build of the PyTorch Live CLI. You can use the nightly by adding `@nightly` after the `torchlive-cli`. For example, the `setup-dev` command using the nightly looks as follows:

```
npx torchlive-cli@nightly setup-dev
```

I don't have an M1 device at hand, so I haven't tested the nightly on an M1 device myself (I requested a device today, so hopefully I'll be able to better support M1 in the near future ;)). We are also currently evaluating the option of upgrading PyTorch Live to use React Native v0.65+, which promises to fix several Apple Silicon device issues (e.g., M1/M1 Pro)

**Comment by tzpr** [comment]
Hi @aaronespasa, thank you for the information and help!
I just got the PyTorch Live example project (MyFirstProject) running on my Android 10 phone (rather old Nokia 6.1).

The key thing was that I started running the terminal with Rosetta. I also realized that I needed to install Homebrew using Rosetta because it installs in different location than the M1 version that I already had.
I went through all the steps listed above using Rosetta and `arch -x86_64` Homebrew. One thing that is different is that my Python version is 3.9.9., couldn't find 3.9.5 from homebrew.

The `setup-dev` wasn't satisfied with the Android SDK Manager installation and started installing the items. It was taking some time so I ended up checking from the `AndroidSDKManagerInstaller.ts` file what are the items installed in that step and installed them from the terminal using the `sdkmanager` command. This way I was able to see what took time and how long. After that I executed the `npx torchlive-cli@nightly setup-dev`command again.

I was using the nightly version of the `torchlive-cli`like @raedle advised, but like I said I didn't try the emulator yet, just my phone. 

Thanks again for the help and quick reply @aaronespasa and @raedle .
